### PR TITLE
Code style changes, add support for extra endpoints.

### PR DESCRIPTION
**Pull request by SiliconValet**
* Added doc blocks for files, functions.
* Consistent whitespacing / formatting.
* Add rest endpoint for listing patterns.
* Add functionality to render endpoint for web component, fragment and son response.
* Add debouncer to the schema editor to prevent DDOS.

### CODE REVIEW

**Comment by siwinski** [comment]
@SiliconValet  THANK YOU!   Me and/or @andrewmriley will review this as soon as we can -- most likely whenever it gets prioritized for us.

File: src/PatternKit/ApiControllerProvider.php [hunk 1]
```diff
@@ -1,134 +1,237 @@
...
+/**
+ * @file ApiControllerProvider.php
+ */
 
+namespace PatternKit;
 
 use Silex\Application;
+use Symfony\Component\Yaml\Yaml;
 use Silex\ControllerProviderInterface;
+//use Symfony\Component\HttpFoundation\JsonResponse;
```
Comment (SiliconValet) [inline src/PatternKit/ApiControllerProvider.php]: this shouldn't have been commented out. my bad. need to issue a fix later.

**Comment by siwinski** [comment]
@SiliconValet would you mind using [php-cs-fixer](http://cs.sensiolabs.org/) (i.e. `php-cs-fixer fix .`) for PSR-1 and PSR-2 coding standards instead of using Drupal coding standards since this is a PHP project and not a Drupal project?

**Comment by siwinski** [comment]
> @SiliconValet would you mind using php-cs-fixer (i.e. php-cs-fixer fix .) for PSR-1 and PSR-2 coding standards instead of using Drupal coding standards since this is a PHP project and not a Drupal project?

@SiliconValet I just added `php-cs-fixer` as a dependency (#14) so if you merge master and update your Composer dependencies you can just run `composer cs`

**Comment by SiliconValet** [comment]
Testing...

**Comment by SiliconValet** [comment]
Works here. Looks good for merge. I don't know how to run a comprehensive test of all the things you guys use, so if you have something, use it.

**Comment by SiliconValet** [comment]
bump for review/merge

**Comment by siwinski** [comment]
@SiliconValet please remove the webrh library-specific stuff

**Comment by siwinski** [comment]
@SiliconValet why the close?

**Comment by siwinski** [comment]
> @SiliconValet why the close?

Nevermind.... I just saw the new merge request

**Pull request closed without merging**